graphql/queries/getComments.ts

```diff
@@ -0,0 +1,18 @@
+import { gql } from '@apollo/client'
+
+const GET_COMMENTS = gql`
+  query getComments($submissionId: Int!, $fileName: String!) {
```

Comment:
Don't know how you're thinking of doing this, but are you planning on doing a query for each file on each submission?

Why not include the comments in the submissions query to have prisma join the tables, so you get everything in a single query?

Reply:
I need to recheck it but almost every question "why does it look so weird?' can be answered by 'because of the way our diff rendering library works'. 

Getting all comments in one submission query should be better indeed.

Reply:
Come to think about it I'm not sure if getting everything in one query would result in better user experience. 

Case in point: when you're reviewing submission you start with one big GET_APP query which is hidden behind loading spinner. And there is GET_SUBMISSION query which gets all submissions for the given lesson (it's using spinner too). The bigger these queries are the longer end user stares at loading screen. Meanwhile, querying for comments case by case is non-blocking. Git diff renders as it is, when comment query gets back you will get a comment box. 

More importantly, to make comments interactive I'm using polling. So there won't be one big query vs many small queries. It's repeating the same query every second, in this case the one with simple select wins over joins.

Reply:
Biggest issue with that would be content shift as you load the comments, what if you had a slow connection and open a submission with a lot of comments? You'd had a page with content jumping around all over the place.

In my opinion, keep this simple for now, query everything together and don't do polling, update the comments by having the user refresh the page. You can tackle these issues separately **IF** needed in the future. Premature optimization only leads to more complexity.

Reply:
I agree with keeping it simple but in this case simple is most likely means using separate queries. I was trying to make top down passing of comments work but ChallengeMaterial needs to be refactored before anything could be done. It's a mess.

Reply:
Yeah, just change the [`GET_SUBMISSIONS`](https://github.com/garageScript/c0d3-app/blob/master/graphql/queries/getSubmissions.ts) query to include the comments list instead of the single comment field that it has now.

Reply:
OK, everything is coming top down from GetApp and GetSubmissions queries. I'm still not sure if I won't need to revert it back to fetching it submission by submission in the future.

Reply:
You need to change [`GET_SUBMISSIONS`](https://github.com/garageScript/c0d3-app/blob/master/graphql/queries/getSubmissions.ts) to include comments too, `GET_APP` only fetches the logged in user submissions. `GET_SUBMISSIONS` fetches the submissions awaiting review.

You already changed the resolver, only the query that needs to be fixed.

Reply:
Yes, I forgot to update the query. Fixed.